.ci/Jenkinsfile

```
@@ -47,6 +47,11 @@ pipeline {
     stage('Lint') {
       steps {
         cleanup()
+        dir("${BASE_DIR}"){
+          whenTrue(isGitRegionMatch(patterns: [".*/_.*"])) {
+            error('_dev in packages are intended to exist only for development purposes.')
+          }
+        }
```

If you prefer you can filter by the branch_name env variable to apply this change in all the branches:

```suggestion
        dir("${BASE_DIR}"){
          whenTrue(isGitRegionMatch(patterns: [".*/_.*"]) && env.BRANCH_NAME?.equals('snapshot')) {
            error('_dev in packages are intended to exist only for development purposes.')
          }
        }
```

---

Hey @v1v, I was planning on making similar PRs to the `staging` and `production` branches. If I do that, do I still need a branch filter for each of these branches in the respective PRs?

---

My suggestion was related to converge the implementation in the same pipeline to help with the backports. Although I'm not familiar with the branch strategy in this project.

---

I see, thanks. 

I think it makes sense to leave out the branch filter. That would let us backport the same exact changes to each of the branches we want this check to be run on. Unless I'm misunderstanding something about Jenkins pipelines, I _think_ that will give us the desired effect. I'm going to go with this for now and see if it works as expected; if not, I will make follow up PRs to add the branch filter.

Thanks again!